webpack/prod.config.js

```diff
@@ -42,34 +42,76 @@ module.exports = (env, args) => {
   const nvrSettings = settingsObject.settings;
 
   return settingsObject.webpackMerge(baseConfig, {
-    //devtool: 'cheap-module-source-map',
+    devtool: 'cheap-module-source-map',
     module: {
-      rules: [{
-        test: /\.html$/,
-        loader: 'html-loader',
-        query: {
-          minimize: true,
+      rules: [
+        {
+          test: /\.html$/,
+          loader: 'html-loader',
+          query: {
+            minimize: true,
+          },
         },
-      }],
+      ],
     },
     optimization: {
       minimize: true,
+      minimizer: [
+        {
+          apply: (compiler) => {
+            /**
+             * Setup the UglifyJsPlugin as webpack4 does, plus options
+             * we decide to override.
+             */
+            // Lazy load the uglifyjs plugin
+            const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+            new UglifyJsPlugin({
+              cache: true, // webpack4: default
+              parallel: true, // webpack4: default
+              sourceMap:
+                (args.devtool && /source-?map/.test(args.devtool)) ||
+                (args.plugins &&
+                  args.plugins.some(
+                    (p) => p instanceof webpack.SourceMapDevToolPlugin
+                  )),
+              uglifyOptions: {
+                compress: {
+                  drop_console: true, // Remove all console.log etc.
+                  keep_infinity: true, // Do not change to 1/0
+                  warnings: false, // Do not warn when dropping
+                },
+                output: {
+                  // Eliminate most comments, but not marked ones
```

Comment:
Why keep these? and what's "marked"? is this the default `/@preserve|@license|@cc_on/i` mentioned [here](http://lisperator.net/uglifyjs/codegen)?

Reply:
still wondering about this

Reply:
I mean, I'm still wondering about the question I asked before:

> Why keep these? and what's "marked"? is this the default `/@preserve|@license|@cc_on/i` mentioned [here](http://lisperator.net/uglifyjs/codegen)?

Reply:
Hmm. Did not get your earlier comments in email notifications so did not see it when I replied. 

Yes: Marked means it uses the regex "/@preserve|@license|@cc_on/" applied within multi-line comments.

The reason to keep these is because when we, or third party code like jQuery, mark things this way, we are bound to keep the comments.